Add documentation for Diyanet integration (#42473)
home-assistant/home-assistant.github.io: source/_integrations/diyanet.markdown

---
title: Diyanet Awqat Salah
description: Instructions on how to integrate Diyanet Awqat Salah API within Home Assistant.
ha_category:
  - Sensor

> ha_release: 2025.12

Reviewer: ```suggestion
ha_release: 2026.1
```

ha_iot_class: Cloud Polling
ha_config_flow: true
ha_codeowners:
  - '@mahdemir'
ha_domain: diyanet
ha_platforms:
  - sensor
ha_integration_type: service
ha_quality_scale: bronze
---

> Use the Diyanet integration to fetch daily Islamic prayer times from the [Diyanet Awqat Salah API](https://awqatsalah.diyanet.gov.tr/index.html) for your selected city. The integration authenticates with your Diyanet account, keeps the current day's times cached, and refreshes them automatically each day.

Reviewer: ```suggestion
The **Diyanet** {% term integration %} retrieves the daily Islamic prayer times from the [Diyanet Awqat Salah API](https://awqatsalah.diyanet.gov.tr/index.html) for your selected city. The integration authenticates with your Diyanet account, keeps the current day's times cached, and refreshes them automatically each day.
```

## Prerequisites

Before setting up the Diyanet integration, ensure you have:

1. Have a Diyanet Awqat Salah account (email/password). If you don’t have one, fill the [API access form](https://awqatsalah.diyanet.gov.tr/files/b8ed95e5-95fc-4635-9c40-406caeeddc1d.pdf) and email it to [dinisleriyk@diyanet.gov.tr](mailto:dinisleriyk@diyanet.gov.tr).

{% include integrations/config_flow.md %}

{% configuration_basic %}

> E-mail:

Author: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/home-assistant/home-assistant.io/pull/42473#discussion_r2602159693)

  description: "Enter the email address associated with your Diyanet Awqat Salah API account. You can find this in the confirmation email you received after registering for API access."
Password:
  description: "Enter the password for your Diyanet Awqat Salah API account. This is the password you set when registering for API access."
{% endconfiguration_basic %}

After signing in, the flow guides you through selecting a country, state, and city. The integration stores the chosen location ID and schedules a daily refresh.

## Supported functionality

The integration adds the following sensors for your selected city:

> - `Fajr prayer time`

Reviewer: ```suggestion
- **Fajr prayer time**
```
We use [bold to markup UI strings](https://developers.home-assistant.io/docs/documenting/general-style-guide). Please change throughout the page.

- `Sunrise`
- `Dhuhr prayer time`
- `Asr prayer time`
- `Maghrib prayer time`
- `Isha prayer time`
- `Qibla time`
- `Hijri date`
- `Gregorian date`

## Services

- `diyanet.refresh`: Fetch the latest prayer times immediately. Provide `config_entry_id` to refresh a specific entry or leave it empty to refresh all loaded Diyanet entries.

## Data updates

Prayer times are refreshed daily and cached for the current day. Use the refresh service if you need an immediate update (for example after changing your location in Diyanet).

> ## Known issues

Author: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/home-assistant/home-assistant.io/pull/42473#discussion_r2602159881)
